src/NeverDestroyed_TEST.cc

```diff
@@ -0,0 +1,95 @@
+#include <gtest/gtest.h>
+
+#include <random>
+#include <unordered_map>
+
+#include "ignition/utils/NeverDestroyed.hh"
+
+class Boom : public std::exception { };
+struct DtorGoesBoom {
+  ~DtorGoesBoom() noexcept(false) { throw Boom(); }
```

Comment:
I don't know any other way to fix MSVC the warning other than suppressing it:

```suggestion
#ifdef _WIN32
// Disable warning C4722 which is triggered by the exception thrown in the dtor
#pragma warning(push)
#pragma warning(disable : 4722)
#endif
  ~DtorGoesBoom() noexcept(false) { throw Boom(); }
#ifdef _WIN32
#pragma warning(pop)
#endif
```